integration-tests/chain-signatures/tests/cases/mod.rs

```diff
@@ -320,3 +320,31 @@ async fn test_multichain_reshare_with_lake_congestion() -> anyhow::Result<()> {
     })
     .await
 }
+
+#[test(tokio::test)]
+async fn test_multichain_reshare_with_first_node_leave() -> anyhow::Result<()> {
```

Comment:
Should we extend the existing test with one more resharing? Not to make are test runs longer.

Reply:
Sounds good

Reply:
Not possible to expand the remove last participant test due to #699 .  With 3 nodes, removing and add a new one will fail due to #699, so can only remove one participant in a test. With 5 nodes, threshold 3, remove two nodes test passed locally and should work on nightly, but it takes too much ram to run on regular ci. I'll merge these tests to one after figure out #699